kernel/sucompat.c

```diff
@@ -196,6 +197,21 @@ static int execve_handler_pre(struct kprobe *p, struct pt_regs *regs)
 	return ksu_handle_execveat_sucompat(fd, filename_ptr, NULL, NULL, NULL);
 }
 
+static int sys_execve_handler_pre(struct kprobe *p, struct pt_regs *regs)
+{
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 16, 0)
+	struct pt_regs *real_regs = (struct pt_regs *)PT_REGS_PARM1(regs);
+#else
+	struct pt_regs *real_regs = regs;
+#endif
+	const char __user *filename_user =
+		(const char *)PT_REGS_PARM1(real_regs);
+	struct filename *filename_mid = getname(filename_user);
```

Comment:
`getname` may cause the CPU to sleep due to memory allocation, which is not allowed in kprobe handlers; this usage may lead to random kernel panic. You need to manually copy the memory using the `_nofault` series of functions and then directly compare the strings; reference: ksu_handle_faccessat.